neo/Prompt/Commands/WalletAddress.py

```diff
@@ -268,16 +269,18 @@ def SplitUnspentCoin(wallet, asset_id, from_addr, index, divisions, fee=Fixed8.Z
     try:
         passwd = prompt("[Password]> ", is_password=True)
     except KeyboardInterrupt:
-            print("Splitting cancelled")
-            return
+        print("Splitting cancelled")
+        return
     if not wallet.ValidatePassword(passwd):
         print("incorrect password")
         return
 
     if ctx.Completed:
         contract_tx.scripts = ctx.GetScripts()
 
-        relayed = NodeLeader.Instance().Relay(contract_tx)
+        nodemgr = NodeManager()
+        # this blocks, consider moving this wallet function to async instead
```

Comment:
Is this instance of relaying somehow different than the others? I am just wondering about the comment and this is the first time I've seen it.

Reply:
After finishing reading through the changes, I saw one more similar comment as a "TODO". Is this something that should be improved before release?

Reply:
this has todo with slowly migrating to `asyncio`. `relay` in
https://github.com/CityOfZion/neo-python/blob/fa0cfb916c31fa4b6d986d03d376a8cf45f66b4c/neo/Prompt/Commands/WalletAddress.py#L283
should ideally be non-blocking. But because we're calling it from a function that is not defined as `async def` we cannot turn `def relay()` into `async def relay` and `await` it. Therefore relay looks like

https://github.com/CityOfZion/neo-python/blob/fa0cfb916c31fa4b6d986d03d376a8cf45f66b4c/neo/Network/neonetwork/network/nodemanager.py#L282-L289

where `wait_for` is a helper function that can call "awaitable" code from non async code.
It does not have to be improved before release. There is too much code still to port to async. That's why this `wait_for` exists to help in migrating over time. The advice is to only use `asyncio` when starting a new project, but there are cases like this where you'll have to slowly migrate. I got this `wait_for` trick from a guy at Instagram where they're also slowly migrating their code base to asyncio.

Reply:
Nice! Okay, sounds good.